diff -Naur gnomeicu-0.98.2/src/chat.c gnomeicu-0.98.2-msgevent/src/chat.c
--- gnomeicu-0.98.2/src/chat.c	Sat Jan 12 12:02:16 2002
+++ gnomeicu-0.98.2-msgevent/src/chat.c	Sat May 11 23:23:26 2002
@@ -472,7 +472,7 @@
 	return TRUE;
     case 'g': /* Ctrl-G is BEEP */
 	if (ev->state & GDK_CONTROL_MASK) {
-	    gnomeicu_event(EV_CHATBEEP, 0);
+	    gnomeicu_event(EV_CHATBEEP, 0, NULL);
 
 	    c = 0x07;
 	    break;
@@ -1436,7 +1436,7 @@
 	  ccontact->away = -1;
 	  break;
      case 0x07: /* Beep */
-	  gnomeicu_event( EV_CHATBEEP , 0 );
+	  gnomeicu_event( EV_CHATBEEP , 0, NULL );
 	  break;
      case 0x08: /* Backspace */
 	  DelCharacter(ccontact);
diff -Naur gnomeicu-0.98.2/src/events.c gnomeicu-0.98.2-msgevent/src/events.c
--- gnomeicu-0.98.2/src/events.c	Wed May 29 02:45:57 2002
+++ gnomeicu-0.98.2-msgevent/src/events.c	Sat May 11 23:55:30 2002
@@ -10,7 +10,7 @@
 #include "util.h"
 
 /*** Global functions ***/
-void gnomeicu_event(ICQEvType event, UIN_T uin)
+void gnomeicu_event(ICQEvType event, UIN_T uin, gchar *extra)
 {
 	GSList *contact;
 	gboolean enable_sound = FALSE;
@@ -23,7 +23,7 @@
 	{
 		contact = Find_User( uin );
 		if( contact )
-			gnomeicu_fire_external_program(event, kontakt);
+			gnomeicu_fire_external_program(event, kontakt, extra);
 	}
 
 	if( toggles->sound_onlyonline == FALSE ||
@@ -82,7 +82,7 @@
 #endif
 }
 
-void gnomeicu_fire_external_program( ICQEvType event, CONTACT_PTR contact )
+void gnomeicu_fire_external_program( ICQEvType event, CONTACT_PTR contact, gchar *extra )
 {
 	char *command=NULL;
 	gchar *arg0, *arg1, *arg2, *arg3, *arg4;
@@ -194,7 +194,7 @@
 			else
 				arg4 = g_strdup( _("N/A") );
 
-			execlp (command, arg0, arg1, arg2, arg3, arg4, NULL);
+			execlp (command, arg0, arg1, arg2, arg3, arg4, extra, NULL);
 			_exit(1);
         }
         else
diff -Naur gnomeicu-0.98.2/src/events.h gnomeicu-0.98.2-msgevent/src/events.h
--- gnomeicu-0.98.2/src/events.h	Thu Mar 30 05:14:07 2000
+++ gnomeicu-0.98.2-msgevent/src/events.h	Sat May 11 23:31:52 2002
@@ -24,7 +24,7 @@
   EV_CONTLIST   /* Received Contact List */
 } ICQEvType;
 
-void gnomeicu_event( ICQEvType event, UIN_T uin );
-void gnomeicu_fire_external_program( ICQEvType event, CONTACT_PTR contact );
+void gnomeicu_event( ICQEvType event, UIN_T uin, gchar *extra );
+void gnomeicu_fire_external_program( ICQEvType event, CONTACT_PTR contact, gchar *extra );
 
 #endif /* __ICQ_EVENTS_H__ */
diff -Naur gnomeicu-0.98.2/src/response.c gnomeicu-0.98.2-msgevent/src/response.c
--- gnomeicu-0.98.2/src/response.c	Thu Feb 14 00:58:27 2002
+++ gnomeicu-0.98.2-msgevent/src/response.c	Sat May 11 23:19:41 2002
@@ -247,7 +247,7 @@
 			applet_update( Current_Status, FALSE );
 		Show_Quick_Status();
 
-		gnomeicu_event( EV_CHATREQ,uin );
+		gnomeicu_event( EV_CHATREQ,uin,NULL );
 	}
 
 	return ccontact;
@@ -306,7 +306,7 @@
 			applet_update( Current_Status, FALSE );
 		Show_Quick_Status();
 
-		gnomeicu_event( EV_FILEREQ,uin );
+		gnomeicu_event( EV_FILEREQ,uin,NULL );
 	}
 
 	return contact;
@@ -349,7 +349,7 @@
 */
       kontakt->need_update = TRUE;
 
-      gnomeicu_event( EV_MSGRECV, uin );
+      gnomeicu_event( EV_MSGRECV, uin, msg );
 
       history_add_incoming( uin, msg, &msg_time );
 
@@ -429,7 +429,7 @@
     
     kontakt->need_update = TRUE;
 
-    gnomeicu_event( EV_URLRECV,uin );
+    gnomeicu_event( EV_URLRECV,uin,url);
 
     /* Add to personal history file */
 
@@ -495,7 +495,7 @@
   {
     kontakt->need_update = TRUE;
 
-    gnomeicu_event( EV_CONTLIST, uin );
+    gnomeicu_event( EV_CONTLIST, uin, NULL );
 
     new_stored_message = g_new0( stored_message, 1 );
     new_stored_message->type = MESSAGE_CONT_LIST;
@@ -525,7 +525,7 @@
     
     kontakt->need_update = TRUE;
 
-    gnomeicu_event( EV_LISTADD,uin );
+    gnomeicu_event( EV_LISTADD,uin,NULL );
 
     history_add_incoming( uin, "User added you to contact list", &msg_time );
   
@@ -561,7 +561,7 @@
     
     kontakt->need_update = TRUE;
 
-    gnomeicu_event( EV_AUTHREQ,uin );
+    gnomeicu_event( EV_AUTHREQ,uin, NULL );
 
     tempmsg = g_strdup_printf(_("User asked for authorisation:\n%s"), msg);
     history_add_incoming( uin, tempmsg, &msg_time );
diff -Naur gnomeicu-0.98.2/src/showlist.c gnomeicu-0.98.2-msgevent/src/showlist.c
--- gnomeicu-0.98.2/src/showlist.c	Mon Feb 18 12:49:20 2002
+++ gnomeicu-0.98.2-msgevent/src/showlist.c	Sat May 11 23:20:26 2002
@@ -183,7 +183,7 @@
 					OfflineList = g_slist_remove( OfflineList, kontakt );
 					/* If they're not ignored, play a sound */
 					if (kontakt->ignore_list == FALSE && kontakt->confirmed == TRUE) {
-						gnomeicu_event (EV_USERON, kontakt->uin);
+						gnomeicu_event (EV_USERON, kontakt->uin, NULL);
 					}
 					/* If they're a member of our list, */
 					/* our online list has changed      */
@@ -232,7 +232,7 @@
 				{
 					OnlineList = g_slist_remove( OnlineList, kontakt );
 					if (kontakt->ignore_list == FALSE && kontakt->confirmed == TRUE) {
-						gnomeicu_event (EV_USEROFF, kontakt->uin);
+						gnomeicu_event (EV_USEROFF, kontakt->uin, NULL);
 					}
 				}
 
